browser/ui/views/tabs/brave_tab_group_underline.cc

```diff
@@ -26,52 +39,63 @@ void BraveTabGroupUnderline::UpdateBounds(views::View* leading_view,
   gfx::RectF trailing_bounds = gfx::RectF(trailing_view->bounds());
   ConvertRectToTarget(trailing_view->parent(), parent(), &trailing_bounds);
   trailing_bounds.Inset(gfx::InsetsF(GetInsetsForUnderline(trailing_view)));
+  if (trailing_bounds.height()) {
+    trailing_bounds.set_height(trailing_bounds.height() +
+                               BraveTabGroupHeader::kPaddingForGroup);
+  }
 
   gfx::Rect group_bounds = ToEnclosingRect(leading_bounds);
-  group_bounds.UnionEvenIfEmpty(ToEnclosingRect(trailing_bounds));
+  group_bounds.Union(ToEnclosingRect(trailing_bounds));
   if (group_bounds.height() == 0) {
     SetVisible(false);
     return;
   }
 
-  SetBounds(0, group_bounds.y(), kStrokeThickness, group_bounds.height());
+  SetBounds(0, group_bounds.y(),
+            std::max(leading_view->width(), trailing_view->width()),
+            group_bounds.height());
 }
 
 gfx::Insets BraveTabGroupUnderline::GetInsetsForUnderline(
     views::View* sibling_view) const {
   if (!tabs::features::ShouldShowVerticalTabs(tab_group_views_->GetBrowser()))
     return TabGroupUnderline::GetInsetsForUnderline(sibling_view);
 
-  constexpr int kStrokeInsetForVerticalTabs = 4;
-  return gfx::Insets::VH(kStrokeInsetForVerticalTabs, 0);
+  return {};
 }
 
 SkPath BraveTabGroupUnderline::GetPath() const {
   if (!tabs::features::ShouldShowVerticalTabs(tab_group_views_->GetBrowser()))
     return TabGroupUnderline::GetPath();
 
-  // In vertical tabs, underline is not actually "underline'. It's vertical line
-  // at the left side of the tab group. And it has half rounded corners.
-  //
-  //  +  group header   | '+' is the underline.
-  // ++  tab 1          | Drawing starts from top-right and goes
-  // ++  tab 2          | counter-clockwise
-  //  +  tab 3          |
-  //
+  // In vertical tabs, underline is not actually "underline'. It's a enclosing
+  // rounded rect for views in the group.
```

Comment:
It would be nice to move this comment to the header file right above the class decls.

Reply:
Yeah, that'd be better. Done.